src/DynamoCoreWpf/Views/Core/DynamoView.xaml.cs

```diff
@@ -2027,6 +2027,20 @@ private void WindowClosed(object sender, EventArgs e)
             {
                 PerformShutdownSequenceOnViewModel();
             }
+            
+            // Force application shutdown when dynamo window is closed, to prevent process hanging
+            if (!DynamoModel.IsTestMode && Application.Current != null)
+            {
+                try
+                {
+                    Application.Current.Shutdown();
```

Comment:
Would this be host application in certain cases instead of sandbox?

Reply:
yeah i need to test this in Revit. Also, Made it a draft to run some tests and get review.